src/sentry/tasks/relay.py

```diff
@@ -153,4 +107,167 @@ def schedule_update_config_cache(
+            keys = []
+
+    else:
+        assert False
+
+    return keys
+
+
+def compute_project_configs(project_keys):
+    """Computes the project configs for all given project keys."""
+    from sentry.models import ProjectKeyStatus
+    from sentry.relay.config import get_project_config
+
+    config_cache = {}
+    for key in project_keys:
+        if key.status != ProjectKeyStatus.ACTIVE:
+            project_config = {"disabled": True}
+        else:
+            project_config = get_project_config(
+                key.project, project_keys=[key], full_config=True
+            ).to_dict()
+        config_cache[key.public_key] = project_config
+
+    projectconfig_cache.set_many(config_cache)
+
+
+@instrumented_task(
+    name="sentry.tasks.relay.invalidate_project_config",
+    queue="relay_config",
+    acks_late=True,
+    soft_time_limit=30,
+    time_limit=32,
+)
+def invalidate_project_config(
+    organization_id=None, project_id=None, public_key=None, trigger="invalidated", **kwargs
+):
+    """Task which re-computes an invalidated project config.
+
+    This task can be scheduled regardless of whether the :func:`update_config_cache` task is
+    scheduled as well.  It is designed to make sure a new project config is computed if
+    scheduled on an invalidation trigger.  Use :func:`schedule_invalidation_task` to
+    schedule this task as that will take care of the queueing semantics.
+
+    The current implementation has some limitations:
+
+    - The task does not synchronise with the :func:`update_config_cache` task so depending
+      on when tasks complete they might have race conditions on writing the config to the
+      cache and an older version might be written.
+
+    - The task does not synchronise with more recent invocations of itself.
+
+    These will be addressed in the future using config revisions tracked in Redis.
+    """
+    validate_args(organization_id, project_id, public_key)
+    sentry_sdk.set_tag("trigger", trigger)
+
+    # Make sure we start by deleting out deduplication key so that new invalidation triggers
+    # can schedule a new message while we already started computing the project config.
+    projectconfig_debounce_cache.invalidation.mark_task_done(
+        organization_id=organization_id, project_id=project_id, public_key=public_key
+    )
+
+    keys = project_keys_to_update(
+        organization_id=organization_id, project_id=project_id, public_key=public_key
+    )
+    if not keys:
+        return
+
+    if organization_id:
+        # Previous incarnations of this task only delete all the affected configs in this
+        # case, relying of lazily filling them back in as they are requested.  Probably
+        # because some organizations can have thousands of projects and they may not all be
+        # active.  Do the same for now, but this could be improved.
+        projectconfig_cache.delete_many(key.public_key for key in keys)
```

Comment:
Do we want to still keep this behavior? I'm ok with it, but it isn't what we agreed on in the design. I'll update the doc if we decide to keep it.

Reply:
I would keep it. Changing this now might lead to new behavior that we cannot explain.

Reply:
Hum, it seems this detail was not thought of in the design.  I'd also not change this for now but do update the doc to say this would be a good improvement.

Reply:
Doc updated.